physical/azure/azure.go

```diff
@@ -99,19 +99,37 @@ func NewAzureBackend(conf map[string]string, logger log.Logger) (physical.Backen
 		}
 	}
 
-	client, err := storage.NewBasicClientOnSovereignCloud(accountName, accountKey, environment)
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
 	if err != nil {
 		return nil, errwrap.Wrapf("failed to create Azure client: {{err}}", err)
 	}
-	client.HTTPClient = cleanhttp.DefaultPooledClient()
 
-	blobClient := client.GetBlobService()
-	container := blobClient.GetContainerReference(name)
-	_, err = container.CreateIfNotExists(&storage.CreateContainerOptions{
-		Access: storage.ContainerAccessTypePrivate,
-	})
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.%s/%s", accountName, environment.StorageEndpointSuffix, name))
 	if err != nil {
-		return nil, errwrap.Wrapf(fmt.Sprintf("failed to create %q container: {{err}}", name), err)
+		return nil, errwrap.Wrapf("failed to create Azure client: {{err}}", err)
+	}
+
+	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+
+	// TODO which context?
+	ctx := context.Background()
```

Comment:
Let's use a context with a timeout of 5s, similar to [etcd3's](https://github.com/hashicorp/vault/blob/170521481d58961761bfd6877221b005cf0f4c7e/physical/etcd/etcd3.go#L130) behavior.

Reply:
Okay I'll update it but won't make it configurable for the time being